### PR TITLE
Add the first version of whoisactive project

### PR DESCRIPTION
**Pull request by AbbiePetcht**
image

image

### CODE REVIEW

File: samples/sp_whoIsActive/tasks/packagetasks.js [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    vsceArgs.push(packageName);
+
+    // Put it all together and execute the command
+    let command = vsceArgs.join(' ');
+    console.log(command);
+    return exec(command);
+};
+
+
+// function getServiceInstaller() {
```
Comment (kevcunnane) [inline samples/sp_whoIsActive/tasks/packagetasks.js]: Can you remove the commented out code?

File: samples/sp_whoIsActive/tasks/packagetasks.js [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+function getOnlinePackageName() {
+    return packageVals.name + "-" + packageVals.version + ".vsix";
+}
+
+function getOnlinePackagePath() {
+    return './' + getOnlinePackageName();
+}
+
+// TEMPORARY GULP TASKS ////////////////////////////////////////////////////
+gulp.task('package:online', gulp.series("build", function() {
```
Comment (kevcunnane) [inline samples/sp_whoIsActive/tasks/packagetasks.js]: Do you need these? There's no backend service to package right?

File: samples/sp_whoIsActive/tasks/config.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+var path = require('path');
+
+var projectRoot = path.resolve(path.dirname(__dirname));
+var srcRoot = path.resolve(projectRoot, 'src');
+var viewsRoot = path.resolve(srcRoot, 'views');
+var htmlcontentRoot = path.resolve(viewsRoot, 'htmlcontent');
+var outRoot = path.resolve(projectRoot, 'out');
+var htmloutroot = path.resolve(outRoot, 'src/views/htmlcontent');
```
Comment (kevcunnane) [inline samples/sp_whoIsActive/tasks/config.js]: You don't have htmlContent do you? Can you clean up the stuff from the mssql extension that's no longer relevant?

File: samples/sp_whoIsActive/src/sql/cpuDelta.sql [hunk 4]
```diff
@@ -0,0 +1,6 @@
+EXEC sp_WhoIsActive
```
Comment (kevcunnane) [inline samples/sp_whoIsActive/src/sql/cpuDelta.sql]: For all these queries I've found it's good to check for existence of the proc and if it doesn't exist, output results that indicate this. That way thing don't look "broken" before someone executes the "create sqp_whoisactive" task. doing an `IF EXISTS ... ELSE` works wonders for perceived reliability

File: samples/sp_whoIsActive/package.json [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+        "@types/mocha": "^2.2.42",
+        "@types/node": "^7.0.43",
+        "child-process-promise": "^2.2.1",
+        "del": "^3.0.0",
+        "gulp": "^4.0.0",
+        "gulp-color": "0.0.1",
+        "gulp-sourcemaps": "^2.6.4",
+        "gulp-tslint": "^6.0.2",
+        "gulp-typescript": "^3.2.4",
+        "pm-mocha-jenkins-reporter": "^0.2.6",
```
Comment (kevcunnane) [inline samples/sp_whoIsActive/package.json]: Are all these necessary? Given lack of tests you can definitely remove the jenkins-reporter. I'd recommend cleaning these up if possible.

File: samples/sp_whoIsActive/package.json [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+        "should": "^13.2.1",
+        "temp-write": "^3.4.0",
+        "tslint": "^3.14.0",
+        "typemoq": "^2.1.0",
+        "typescript": "^2.6.1",
+        "vscode": "^1.1.6",
+        "vsce": "1.36.2",
+        "vso-node-api": "6.1.2-preview"
```
Comment (kevcunnane) [inline samples/sp_whoIsActive/package.json]: Pretty sure you don't need this either.

File: samples/sp_whoIsActive/src/sql/who_is_active_v11_30.sql [hunk 7]
```diff
@@ -0,0 +1,5273 @@
+SET QUOTED_IDENTIFIER ON;
```
Comment (kevcunnane) [inline samples/sp_whoIsActive/src/sql/who_is_active_v11_30.sql]: Please ensure we have registered this in the OSS tool. Will we be shipping this extension? If so, please ensure we have a thirdPartyNotices file that includes the sp_whoisactive license as a reference.

Comment (AbbiePetcht) [reply samples/sp_whoIsActive/src/sql/who_is_active_v11_30.sql]: Will discuss offline whether we will ship this extension or only provide as sample

---
In reply to: [175923723](https://github.com/Microsoft/sqlopsstudio/pull/943#discussion_r175923723) [](ancestors = 175923723)

**Pull request merged**